ee/query-service/anomaly/params.go

```diff
@@ -0,0 +1,191 @@
+package anomaly
+
+import (
+	"math"
+	"time"
+
+	"go.signoz.io/signoz/pkg/query-service/common"
+	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
+)
+
+type Seasonality string
+
+const (
+	SeasonalityHourly Seasonality = "hourly"
+	SeasonalityDaily  Seasonality = "daily"
+	SeasonalityWeekly Seasonality = "weekly"
+)
+
+func (s Seasonality) IsValid() bool {
+	switch s {
+	case SeasonalityHourly, SeasonalityDaily, SeasonalityWeekly:
+		return true
+	default:
+		return false
+	}
+}
+
+type GetAnomaliesRequest struct {
+	Params      *v3.QueryRangeParamsV3
+	Seasonality Seasonality
+}
+
+type GetAnomaliesResponse struct {
+	Results []*v3.Result
+}
+
+// anomalyParams is the params for anomaly detection
+// prediction = avg(past_period_query) + avg(current_season_query) - avg(past_season_query)
```

Comment:
nit : break it down to maybe to This 
seasonalGrowth := avg(current_season_query) - avg(past_season_query)
prediction = avg(past_period_query)





----

Also did you think about using median instead of avg ? 
![image](https://github.com/user-attachments/assets/9bcece7f-ccd5-4432-b335-123926e6e873)

Reply:
no we are going to use the mean for now. the median with a skewed result will drastically affect results

Reply:
Okay, cool. If it requires we can make it configurable in future if some users demand it.